glib-macros/tests/props.rs

```diff
@@ -0,0 +1,57 @@
+use glib::prelude::*;
+#[test]
+fn props() {
+    mod foo {
+        use glib::prelude::*;
+        use glib::subclass::prelude::*;
+        use glib_macros::Props;
+        use std::cell::RefCell;
+
+        mod imp {
+            use super::*;
+
+            #[derive(Props, Default)]
+            pub struct Foo {
+                #[prop("bar", "bar", "This is a bar", None, glib::ParamFlags::READWRITE)]
+                bar: RefCell<String>,
+                #[prop("buzz", "buzz", "This is a buzz", 1, 100, 1, glib::ParamFlags::READWRITE)]
+                buzz: RefCell<u32>,
+            }
+
+            #[glib::object_subclass]
+            impl ObjectSubclass for Foo {
+                const NAME: &'static str = "MyFoo";
+                type Type = super::Foo;
+            }
```

Comment:
What happens if you want to override ObjectImpl?

Reply:
I think a way to work around this is to move the properties specific stuff to a separate trait and make it a requirement to implement ObjectImpl. If we ever do that, we might want to do the same thing for signals as well to avoid doing yet another API break once there is some kind of macro to generate signals code.

Reply:
Also it should be possible to manually implement the property/signal related methods on `ObjectImpl`. We shouldn't make the macro mandatory as it won't be able to cover all possible use-cases.

Reply:
@sdroege What do you think about what I suggested above with splitting the properties/signals into separate traits this way you can still auto-implement them without conflicting with a manual ObjectImpl implementation?

Reply:
Yes, that's what I also suggested in my proposal some months ago.

Reply:
> I suggested above with splitting the properties/signals...
> I also suggested in my proposal some months ago.

Can you guys provide me with some links about this? I can't find anything.

I've tried splitting the traits, but I haven't found a satisfying solution

Reply:
> I've tried splitting the traits, but I haven't found a satisfying solution 

Maybe push your WIP changes to a branch and we can have a look at it? What I was suggesting, is to make a new PropertiesImpl trait that would have the properties related methods in ObjectImpl and make it a requirement of ObjectImpl trait implementation.

Reply:
Ok, I'll send another WIP PR with only those changes.